data/lua/wml-tags.lua

```diff
@@ -141,7 +141,9 @@ function wml_actions.fire_event(cfg)
 	local w2 = helper.get_child(cfg, "secondary_attack")
 	if w2 then w1 = w1 or {} end
 
-	wesnoth.fire_event(cfg.name, x1, y1, x2, y2, w1, w2)
+	if cfg.id and cfg.id ~= "" then wesnoth.fire_event_by_id(cfg.id, x1, y1, x2, y2, w1, w2)
+	elseif cfg.name and cfg.name ~= "" then wesnoth.fire_event_by_name(cfg.name, x1, y1, x2, y2, w1, w2)
+	end
```

Comment:
I would read a [fire_event] with both id= and name= as meaning an AND relationship. Since the id should be unique, the name= is basically unneeded. Specifying both either works the same omitting the name= or fails. It might be a good idea to debug-level warn on this failure.

I would suggest an event without a name would now be valid. It would be a custom event, of course. And it ensures there can only ever one handler for the fire_event; there would be no way to hang a second handler on it due to the unique-id constraint.